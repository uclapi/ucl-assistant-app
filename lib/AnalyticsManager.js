import * as Amplitude from 'expo-analytics-amplitude'
import Constants from 'expo-constants'

const AMPLITUDE_API_KEY = Constants.manifest.extra
  && Constants.manifest.extra.amplitude
  && Constants.manifest.extra.amplitude.apiKey
  ? Constants.manifest.extra.amplitude.apiKey
  : null

const initialise = () => Amplitude.initialize(AMPLITUDE_API_KEY)

const setUserId = (userId) => {
  // make sure this is unique
  if (userId) {
    Amplitude.setUserId(userId)
  }
}

const setUserProperties = ({ ...userProperties }) => {
  Amplitude.setUserProperties({ ...userProperties })
}

const clearUserProperties = () => {
  Amplitude.clearUserProperties()
}

const logEvent = (eventName, eventProperties) => {
  if (eventProperties) {
    Amplitude.logEventWithProperties(eventName, eventProperties)
  } else {
    Amplitude.logEvent(eventName)
  }
}

const logScreenView = (screenName) => logEvent(`VIEW_SCREEN_${screenName}`)

const events = {
  NOTIFICATIONS_ENABLE: `NOTIFICATIONS_ENABLE`,
  NOTIFICATIONS_SKIP: `NOTIFICATIONS_SKIP`,
  PUSH_NOTIFICATIONS_REGISTER: `PUSH_NOTIFICATIONS_REGISTER`,
  SETTINGS_GIVE_FEEDBACK: `SETTINGS_GIVE_FEEDBACK`,
  SETTINGS_RATE_APP: `SETTINGS_RATE_APP`,
  SETTINGS_VIEW_FAQS: `SETTINGS_VIEW_FAQS`,
}

const AnalyticsManager = {
  clearUserProperties,
  events,
  initialise,
  logEvent,
  logScreenView,
  setUserId,
  setUserProperties,
}

export default (AMPLITUDE_API_KEY && !__DEV__)
  ? AnalyticsManager
  : (
    new Proxy({}, {
      get() {
        return () => { } // do nothing in development mode
      },
    })
  )
