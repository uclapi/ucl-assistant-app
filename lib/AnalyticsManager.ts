import * as Amplitude from 'expo-analytics-amplitude'
import Constants from 'expo-constants'

import configureStore from "../configureStore"

const { store } = configureStore

const AMPLITUDE_API_KEY = Constants.manifest.extra
  && Constants.manifest.extra.amplitude
  && Constants.manifest.extra.amplitude.apiKey
  ? Constants.manifest.extra.amplitude.apiKey
  : null

const initialise = (): Promise<void> => Amplitude.initializeAsync(AMPLITUDE_API_KEY)

const shouldTrackAnalytics = (): boolean => (
  AMPLITUDE_API_KEY
  && !__DEV__
  && store.getState().user.settings.shouldTrackAnalytics
)

const setUserId = (userId): void => {
  if (!shouldTrackAnalytics()) { return }

  // make sure this is unique
  if (userId) {
    Amplitude.setUserIdAsync(userId)
  }
}

const setUserProperties = ({ ...userProperties }): void => {
  if (!shouldTrackAnalytics()) { return }

  Amplitude.setUserPropertiesAsync({ ...userProperties })
}

const clearUserProperties = (): void => {
  Amplitude.clearUserPropertiesAsync()
}

const logEvent = (eventName: string, eventProperties?: unknown): void => {
  if (!shouldTrackAnalytics()) { return }

  if (eventProperties) {
    Amplitude.logEventWithPropertiesAsync(eventName, eventProperties)
  } else {
    Amplitude.logEventAsync(eventName)
  }
}

const logScreenView = (screenName: string): void => {
  if (!shouldTrackAnalytics()) { return }

  logEvent(`VIEW_SCREEN_${screenName}`)
}

const events = {
  NOTIFICATIONS_ENABLE: `NOTIFICATIONS_ENABLE`,
  NOTIFICATIONS_SKIP: `NOTIFICATIONS_SKIP`,
  PUSH_NOTIFICATIONS_REGISTER: `PUSH_NOTIFICATIONS_REGISTER`,
  SETTINGS_GIVE_FEEDBACK: `SETTINGS_GIVE_FEEDBACK`,
  SETTINGS_RATE_APP: `SETTINGS_RATE_APP`,
  SETTINGS_VIEW_FAQS: `SETTINGS_VIEW_FAQS`,
}

export const AnalyticsManager = {
  clearUserProperties,
  events,
  initialise,
  logEvent,
  logScreenView,
  setUserId,
  setUserProperties,
}

export default AnalyticsManager
