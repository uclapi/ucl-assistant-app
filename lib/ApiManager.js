import axios from "axios";
import moment from "moment";
import { encode as btoa } from "base-64";
import {
  ROOMS_URL,
  WORKSPACES_URL,
  EQUIPMENT_URL,
  ROOMBOOKINGS_URL,
} from "../constants/API";

const roomsController = {
  search: async (token = null, query) => {
    if (query && query.length < 3) {
      return {};
    }
    try {
      const results = await axios.get(ROOMS_URL, {
        params: {
          query,
        },
        headers: {
          authorization: `Bearer ${token}`,
          "uclapi-roombookings-version": "1",
        },
      });
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error);
      }
      return results.data.content.rooms;
    } catch (error) {
      throw error;
    }
  },
  getEquipment: async (token = null, { roomid, siteid }) => {
    if (!roomid || !siteid) {
      throw new Error("Must specify roomid and siteid");
    }
    try {
      const results = await axios.get(EQUIPMENT_URL, {
        params: {
          roomid,
          siteid,
        },
        headers: {
          authorization: `Bearer ${token}`,
        },
      });
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error);
      }
      return results.data.content.equipment;
    } catch (error) {
      throw error;
    }
  },
  getBookings: async (
    token = null,
    { roomid, siteid, date = moment().format("YYYYMMDD") },
  ) => {
    if (!roomid || !siteid) {
      throw new Error("Must specify roomid and siteid");
    }
    try {
      const results = await axios.get(ROOMBOOKINGS_URL, {
        params: {
          roomid,
          siteid,
          date,
        },
        headers: {
          authorization: `Bearer ${token}`,
        },
      });
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error);
      }
      return results.data.content.bookings;
    } catch (error) {
      throw error;
    }
  },
};

const workspacesApi = axios.create({
  baseURL: WORKSPACES_URL,
});

const workspacesController = {
  getLiveImage: async (token = null, { surveyId, mapId }) => {
    try {
      const result = await workspacesApi.get("/getliveimage/map.svg", {
        params: {
          survey_id: surveyId,
          map_id: mapId,
        },
        headers: {
          authorization: `Bearer ${token}`,
        },
        responseType: "arraybuffer",
      });
      const image = btoa(
        new Uint8Array(result.data).reduce(
          (data, byte) => data + String.fromCharCode(byte),
          "",
        ),
      );
      // there may be no need to base64 SVGs
      // but I have yet to test this extensively
      // https://css-tricks.com/probably-dont-base64-svg/
      return `data:image/svg+xml;base64,${image}`;
    } catch (error) {
      throw error;
    }
  },
};

export default {
  rooms: roomsController,
  workspaces: workspacesController,
};
