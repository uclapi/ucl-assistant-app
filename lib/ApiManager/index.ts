import Constants from 'expo-constants'
import PeopleController from './PeopleController'
import RoomsController from './RoomsController'
import TimetableController from './TimetableController'
import WorkspacesController from './WorkspacesController'

const clientId = Constants.manifest.extra?.uclapi?.clientId

export default {
  people: PeopleController,
  rooms: RoomsController,
  timetable: TimetableController,
  workspaces: WorkspacesController,
  clientId,
}
