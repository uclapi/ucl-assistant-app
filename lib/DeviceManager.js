import * as Device from 'expo-device'

const isRealDevice = () => Device.isRealDevice

export default {
  isRealDevice,
}
