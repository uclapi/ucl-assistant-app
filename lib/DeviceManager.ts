import * as Device from 'expo-device'
import * as Network from 'expo-network'
import { Platform } from 'react-native'

const isWeb = Platform.OS === `web`
const isAndroid = Platform.OS === `android`
const isiOS = Platform.OS === `ios`

const isRealDevice = Device.isDevice

const isConnectedToInternet = async (): Promise<boolean> => {
  if (isAndroid) {
    const isAirplaneMode = await Network.isAirplaneModeEnabledAsync()
    if (isAirplaneMode) {
      return false
    }
  } else {
    return false
  }

  const { isInternetReachable } = await Network.getNetworkStateAsync()
  return isInternetReachable
}

export default {
  isAndroid,
  isConnectedToInternet,
  isRealDevice,
  isWeb,
  isiOS,
}
