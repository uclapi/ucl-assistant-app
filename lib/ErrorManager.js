import Constants from 'expo-constants'
import * as Sentry from "sentry-expo"

const initialise = () => {
  if (!__DEV__) {
    const { extra: { sentry: { dsn } = {} } = {} } = Constants.manifest
    Sentry.init({
      dsn,
    })

    const { revisionId } = Constants.manifest
    if (typeof revisionId === `string` && revisionId.length > 0) {
      Sentry.setRelease(Constants.manifest.revisionId)
    }
  }
}

const captureError = (error) => {
  if (!__DEV__) {
    Sentry.captureException(error)
  } else {
    console.error(error)
  }
}

export default {
  captureError,
  initialise,
}
