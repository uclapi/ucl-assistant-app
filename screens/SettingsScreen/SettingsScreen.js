/* eslint-disable react/no-unused-state */
import React, { Component, Fragment } from "react";
import {
  Alert,
  Platform,
  View,
  Clipboard,
  StyleSheet,
  Linking,
} from "react-native";
import { NavigationActions, StackActions } from "react-navigation";
import PropTypes from "prop-types";
import { Constants, IntentLauncherAndroid } from "expo";
import { connect } from "react-redux";
import {
  TitleText,
  BodyText,
  SubtitleText,
  ButtonText,
} from "../../components/Typography";
import { Page, Horizontal, PaddedIcon } from "../../components/Containers";
import { signOut } from "../../actions/userActions";
import Button, { SmallButton } from "../../components/Button";
import Colors from "../../constants/Colors";
import TextInput from "../../components/Input/TextInput";
import NotificationSwitch from "./NotificationSwitch";
import LiveIndicator from "../../components/LiveIndicator";
import common from "../../styles/common";

const { version } = require("../../package.json");

const styles = StyleSheet.create({
  faqButton: {
    marginBottom: 10,
  },
  notificationSettingsButton: {
    marginTop: 10,
  },
  section: {
    marginBottom: 15,
    marginTop: 15,
  },
  textWithUpperMargin: {
    marginTop: 10,
  },
});

class SettingsScreen extends Component {
  static navigationOptions = {
    header: null,
  };

  static propTypes = {
    signOut: PropTypes.func,
    navigation: PropTypes.shape(),
    state: PropTypes.shape(),
  };

  static defaultProps = {
    signOut: () => {},
    navigation: {},
    state: {},
  };

  static mapStateToProps = state => ({
    state,
  });

  static mapDispatchToProps = dispatch => ({
    signOut: () => dispatch(signOut()),
  });

  static launchNotificationSettings() {
    // note that this will only work on standalone apps
    // because the bundleIdentifier/packageName will
    // only be used when an APK/IPA is built
    if (Platform.OS === "android") {
      IntentLauncherAndroid.startActivityAsync(
        IntentLauncherAndroid.ACTION_APP_NOTIFICATION_SETTINGS,
        {
          "android.provider.extra.APP_PACKAGE":
            Constants.manifest.android.package,
        },
      );
    } else {
      // is iOS
      Linking.openURL(
        `app-settings://notification/${
          Constants.manifest.ios.bundleIdentifier
        }`,
      );
    }
  }

  static releaseChannelStyle = {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
  };

  state = {
    isSigningOut: false,
  };

  componentDidUpdate(_, prevState) {
    if (prevState.isSigningOut && this.props.state.user.token === "") {
      const action = StackActions.reset({
        index: 0,
        actions: [NavigationActions.navigate({ routeName: "Splash" })],
      });
      this.props.navigation.dispatch(action);
    }
  }

  signOut = () => {
    this.props.signOut();
    this.setState({ isSigningOut: true });
  };

  navigateToFAQs = () => {
    this.props.navigation.navigate("FAQ");
  };

  async copyTokenToClipboard() {
    await Clipboard.setString(this.props.state.user.token);
    Alert.alert("Copied", "Token copied to clipboard.");
  }

  render() {
    const { state } = this.props;
    return (
      <Page mainTabPage>
        <TitleText>Settings</TitleText>
        <View>
          <SubtitleText>Notifications</SubtitleText>
          <NotificationSwitch />
          <Button
            onPress={() => SettingsScreen.launchNotificationSettings()}
            style={styles.notificationSettingsButton}
          >
            Manage Notification Settings
          </Button>
        </View>
        <View style={styles.section}>
          <SubtitleText>User</SubtitleText>
          <BodyText>Logged in as {state.user.fullName}</BodyText>
          <BodyText>Unique Person Identifier (UPI): {state.user.upi}</BodyText>
          <Button onPress={this.signOut}>
            <Horizontal>
              <PaddedIcon
                name="log-out"
                size={24}
                color={Colors.pageBackground}
              />
              <ButtonText>Sign Out</ButtonText>
            </Horizontal>
          </Button>
        </View>
        <View style={styles.section}>
          <TitleText>About This App</TitleText>
          <Button onPress={this.navigateToFAQs} style={styles.faqButton}>
            Frequently Asked Questions
          </Button>
          <BodyText>Version {version}</BodyText>
          <Horizontal style={SettingsScreen.releaseChannelStyle}>
            {__DEV__ ? (
              <LiveIndicator>Developer Mode</LiveIndicator>
            ) : (
              <Fragment>
                <BodyText>Release Channel: </BodyText>
                <LiveIndicator>
                  {Constants.manifest.releaseChannel || "dev"}
                </LiveIndicator>
              </Fragment>
            )}
          </Horizontal>
        </View>
        <View style={styles.section}>
          <SubtitleText>Author</SubtitleText>
          <BodyText>
            Created by Matt Bell (class of 2018) using the UCL API.
          </BodyText>
          <BodyText style={styles.textWithUpperMargin}>
            Currently managed by the UCL API Team: a group of students working
            together within ISD to improve UCL by building a platform on top of
            UCL{`'`}s digital services for students.
          </BodyText>
          <BodyText style={styles.textWithUpperMargin}>
            Illustrations courtesy of the unDraw project, released under the MIT
            license.
          </BodyText>
        </View>
        {__DEV__ && (
          <View style={styles.section}>
            <TitleText>Dev Stuff</TitleText>
            <SubtitleText>UCL API Token</SubtitleText>
            <Horizontal>
              <TextInput style={common.flex} value={state.user.token} />
              <SmallButton onPress={() => this.copyTokenToClipboard()}>
                Copy
              </SmallButton>
            </Horizontal>
            <SubtitleText>State</SubtitleText>
            <BodyText>{JSON.stringify(state, "\n", 2)}</BodyText>
          </View>
        )}
      </Page>
    );
  }
}

export default connect(
  SettingsScreen.mapStateToProps,
  SettingsScreen.mapDispatchToProps,
)(SettingsScreen);
