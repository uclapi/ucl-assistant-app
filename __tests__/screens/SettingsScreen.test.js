/**
 * @jest-environment jsdom
 */
import "react-native"

import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'

import { MailManager, WebBrowserManager } from '../../lib'
import { SettingsScreen } from "../../screens/SettingsScreen"

const {
  repository: {
    url: githubURL,
  },
} = require(`../../package.json`)

Enzyme.configure({ adapter: new Adapter() })

describe(`SettingsScreen`, () => {
  let wrapper
  const mockSignOut = jest.fn()
  const mockNavigate = jest.fn()
  const mockSetShouldTrackAnalytics = jest.fn()
  const mockProps = {
    navigation: {
      navigate: mockNavigate,
    },
    setShouldTrackAnalytics: mockSetShouldTrackAnalytics,
    signOut: mockSignOut,
    user: {
      fullName: `Jeremy Bentham`,
      settings: {
        shouldTrackAnalytics: false,
      },
      token: `abc123`,
      upi: `upiupi11`,
    },
  }

  const mockOpenLink = jest.fn(() => Promise.resolve())
  WebBrowserManager.openLink = mockOpenLink

  const mockComposeAsync = jest.fn(() => Promise.resolve())
  MailManager.composeAsync = mockComposeAsync

  beforeEach(() => {
    wrapper = shallow(<SettingsScreen {...mockProps} />)
  })

  it(`renders the SettingsScreen`, () => {
    expect(wrapper).toMatchSnapshot()
  })

  it(`logs out when the signOut button is pressed`, async () => {
    const signOutButton = wrapper.findWhere(
      (node) => node.prop(`testID`) === `signOutButton`,
    )
    expect(signOutButton.exists())
    signOutButton.props().onPress()

    expect(mockSignOut).toHaveBeenCalledTimes(1)
  })

  it(`opens Github repository when Github button is pressed`, async () => {
    const githubButton = wrapper.findWhere(
      (node) => node.prop(`testID`) === `githubButton`,
    )
    expect(githubButton.exists())
    expect(githubButton.prop(`href`) === githubURL)
  })

  it(`opens FAQ page when the FAQ button is pressed`, async () => {
    const faqButton = wrapper.findWhere(
      (node) => node.prop(`testID`) === `faqButton`,
    )
    expect(faqButton.exists())

    faqButton.props().onPress()

    expect(mockNavigate).toHaveBeenCalledTimes(1)
    expect(mockNavigate).toHaveBeenCalledWith(`FAQ`)
  })

  it(`opens mail client when the feedback button is pressed`, async () => {
    const feedbackButton = wrapper.findWhere(
      (node) => node.prop(`testID`) === `feedbackButton`,
    )
    expect(feedbackButton.exists())

    feedbackButton.props().onPress()
    expect(mockComposeAsync).toHaveBeenCalledTimes(1)
    expect(mockComposeAsync.mock.calls).toMatchSnapshot()
  })

  it(`dispatches action when analytics checkbox is toggled`, async () => {
    const analyticsCheckbox = wrapper.findWhere(
      (node) => node.prop(`testID`) === `analyticsCheckbox`,
    )
    expect(analyticsCheckbox.exists())

    analyticsCheckbox.props().onValueChange(true)
    expect(mockSetShouldTrackAnalytics).toHaveBeenCalledTimes(1)
    expect(mockSetShouldTrackAnalytics).toHaveBeenCalledWith(true)
  })
})
