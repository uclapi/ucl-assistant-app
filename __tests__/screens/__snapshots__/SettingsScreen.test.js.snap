// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen opens mail client when the feedback button is pressed 1`] = `
Array [
  Array [
    Object {
      "body": "I've been using UCL Assistant and I just wanted to tell you ... 

Technical Information
{
  \\"deviceName\\": \\"Test Phone\\",
  \\"platform\\": {},
  \\"upi\\": \\"upiupi11\\"
}",
      "recipients": Array [
        "isd.apiteam@ucl.ac.uk",
      ],
      "subject": "Feedback about UCL Assistant",
    },
  ],
]
`;

exports[`SettingsScreen renders the SettingsScreen 1`] = `
<Page
  bottomColour="#fefefe"
  contentContainerStyle={Object {}}
  keyboardAvoidingViewStyle={Object {}}
  mainTabPage={true}
  onRefresh={[Function]}
  refreshEnabled={false}
  refreshing={false}
  safeAreaViewStyle={Object {}}
  style={Object {}}
  topColour="#fefefe"
>
  <View
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 15,
      }
    }
  >
    <HeaderText
      style={Object {}}
    >
      User
    </HeaderText>
    <BodyText
      style={Object {}}
    >
      Logged in as Jeremy Bentham
    </BodyText>
    <BodyText
      style={Object {}}
    >
      Unique Person Identifier (UPI): upiupi11
    </BodyText>
    <Link
      containerStyle={Object {}}
      href=""
      onPress={[Function]}
      style={
        Object {
          "marginTop": 10,
        }
      }
      testID="signOutButton"
    >
      Sign Out
    </Link>
  </View>
  <View
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 15,
      }
    }
  >
    <HeaderText
      style={Object {}}
    >
      App Info
    </HeaderText>
    <Link
      containerStyle={Object {}}
      href=""
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 10,
        }
      }
      testID="faqButton"
    >
      Frequently Asked Questions
    </Link>
    <BodyText
      style={Object {}}
    >
      Version: 5.0.0
    </BodyText>
    <Horizontal
      bottomColour="#fefefe"
      contentContainerStyle={Object {}}
      keyboardAvoidingViewStyle={Object {}}
      mainTabPage={false}
      onRefresh={[Function]}
      refreshEnabled={false}
      refreshing={false}
      safeAreaViewStyle={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 10,
        }
      }
      topColour="#fefefe"
    >
      <LiveIndicator
        style={Object {}}
      >
        Developer Mode
      </LiveIndicator>
    </Horizontal>
    <Link
      containerStyle={Object {}}
      href="https://github.com/uclapi/ucl-assistant-app"
      onPress={null}
      style={Object {}}
      testID="githubButton"
    >
      Source Code
    </Link>
    <Link
      containerStyle={
        Object {
          "marginVertical": 10,
        }
      }
      href=""
      onPress={[Function]}
      style={Object {}}
      testID="feedbackButton"
    >
      Send Us Feedback
    </Link>
  </View>
  <View
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 15,
      }
    }
  >
    <HeaderText
      style={Object {}}
    >
      Credits
    </HeaderText>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <BodyText
        style={Object {}}
      >
        Created by Matt Bell (class of 2018) using theÂ 
      </BodyText>
      <Link
        containerStyle={Object {}}
        href="https://uclapi.com"
        onPress={null}
        style={Object {}}
      >
        UCL API
      </Link>
      <BodyText
        style={Object {}}
      >
        .
      </BodyText>
    </View>
    <BodyText
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Currently managed by the UCL API Team: a group of students working together within ISD to improve UCL by building a platform on top of UCL's digital services for students.
    </BodyText>
    <BodyText
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Illustrations courtesy of the unDraw project, released under the MIT license.
    </BodyText>
  </View>
  <View
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 15,
      }
    }
  >
    <HeaderText
      style={Object {}}
    >
      UCL API Token
    </HeaderText>
    <Horizontal
      bottomColour="#fefefe"
      contentContainerStyle={Object {}}
      keyboardAvoidingViewStyle={Object {}}
      mainTabPage={false}
      onRefresh={[Function]}
      refreshEnabled={false}
      refreshing={false}
      safeAreaViewStyle={Object {}}
      style={Object {}}
      topColour="#fefefe"
    >
      <TextInputComponent
        allowFontScaling={true}
        rejectResponderTermination={true}
        style={
          Object {
            "flex": 1,
          }
        }
        underlineColorAndroid="transparent"
        value="abc123"
      />
      <SmallButton
        disabled={false}
        loading={false}
        onPress={[Function]}
        style={Object {}}
      >
        Copy
      </SmallButton>
    </Horizontal>
    <HeaderText
      style={Object {}}
    >
      State
    </HeaderText>
    <BodyText
      style={Object {}}
    >
      {
  "fullName": "Jeremy Bentham",
  "token": "abc123",
  "upi": "upiupi11"
}
    </BodyText>
  </View>
</Page>
`;
